### PR TITLE
Add service_account_credentials for Google Cloud and Drive

### PR DESCRIPTION
**Pull request by mholt**
Adds the ability to configure a service account without needing a file on disk: `service_account_credentials` (the contents of the service account credentials file) can be put directly into an environment variable. This works for Google Cloud Storage and Google Drive!

As discussed here: https://forum.rclone.org/t/accessing-google-cloud-storage-without-service-account-file/5461/5?u=matt

Includes a mention in the docs.

Thanks again for the pointers. Let me know if there's anything else!

### CODE REVIEW

File: backend/drive/drive.go [hunk 1]
```diff
@@ -466,9 +462,18 @@ func createOAuthClient(name string) (*http.Client, error) {
...
-	serviceAccountPath := config.FileGet(name, "service_account_file")
-	if serviceAccountPath != "" {
-		oAuthClient, err = getServiceAccountClient(serviceAccountPath)
+	// try loading service account credentials from env variable, then from a file
+	serviceAccountCreds := []byte(config.FileGet(name, "service_account_credentials"))
+	if len(serviceAccountCreds) == 0 {
+		serviceAccountPath := config.FileGet(name, "service_account_file")
+		loadedCreds, err := ioutil.ReadFile(os.ExpandEnv(serviceAccountPath))
+		if err != nil {
+			log.Fatalf("Error opening service account credentials file: %v", err)
```
Comment (ncw) [inline backend/drive/drive.go]: Can you make that an error return please?  I've been trying to rid the code of `log.Fatalf` :-)

```go
return nil, errors.Wrap(err, "error opening credentials file")
```

File: backend/googlecloudstorage/googlecloudstorage.go [hunk 2]
```diff
@@ -295,9 +291,18 @@ func NewFs(name, root string) (fs.Fs, error) {
...
-	serviceAccountPath := config.FileGet(name, "service_account_file")
-	if serviceAccountPath != "" {
-		oAuthClient, err = getServiceAccountClient(serviceAccountPath)
+	// try loading service account credentials from env variable, then from a file
+	serviceAccountCreds := []byte(config.FileGet(name, "service_account_credentials"))
+	if len(serviceAccountCreds) == 0 {
+		serviceAccountPath := config.FileGet(name, "service_account_file")
+		loadedCreds, err := ioutil.ReadFile(os.ExpandEnv(serviceAccountPath))
+		if err != nil {
+			log.Fatalf("Error opening service account credentials file: %v", err)
```
Comment (ncw) [inline backend/googlecloudstorage/googlecloudstorage.go]: And this one too...

File: docs/content/drive.md [hunk 3]
```diff
@@ -202,7 +202,9 @@ actively logged-in users, for example build machines.
 To use a Service Account instead of OAuth2 token flow, enter the path
 to your Service Account credentials at the `service_account_file`
 prompt during `rclone config` and rclone won't use the browser based
-authentication flow.
+authentication flow. If you'd rather stuff the contents of the file
```
Comment (ncw) [inline docs/content/drive.md]: That should probably say "If you'd rather stuff the contents of the credentials file into the rclone config file you can put it under the `service_account_credentials` key, or set the equivalent environment variable."

**Comment by mholt** [comment]
Thanks for the review! Made the relevant changes.

**Comment by ncw** [comment]
Thanks @mholt looks perfect now :-)

**Comment by mholt** [comment]
Woo hoo! Thanks Nick!

**Pull request merged**